x/bridge/observer/bitcoin/bitcoin.go

```diff
@@ -16,170 +17,166 @@ import (
 	"github.com/btcsuite/btcd/rpcclient"
 	"github.com/btcsuite/btcd/txscript"
 	"github.com/cometbft/cometbft/libs/log"
-)
-
-const ModuleNameObserver = "btc-observer"
-
-type TxIn struct {
-	Id          string
-	Height      uint64
-	Sender      string
-	Destination string
-	Amount      math.Uint
-	Memo        string
-}
 
-type RpcConfig struct {
-	Host       string
-	DisableTls bool
-	User       string
-	Pass       string
-}
+	"github.com/osmosis-labs/osmosis/v24/x/bridge/observer"
+)
 
-func (o RpcConfig) Validate() error {
-	if len(o.Host) == 0 {
-		return fmt.Errorf("Invalid `Host` value")
-	} else if len(o.User) == 0 {
-		return fmt.Errorf("Invalid `User` value")
-	} else if len(o.Pass) == 0 {
-		return fmt.Errorf("Invalid `Pass` value")
-	} else {
-		return nil
-	}
-}
+var (
+	ModuleName = "bitcoin-chain"
+)
 
-type Observer struct {
+type Bitcoin struct {
```

Comment:
The change of name feels wrong to me. it will be imported as `bitcoin.Bitcoin` or `bitcoin.NewBitCoin()`. I think observer was pretty clear `bitcoin.NewObserver` is already namespaced in bitcoin context. 
Same for the files name

Reply:
I see, I would be more in favor of ChainClient as it is more representative